network/p2p/p2pbuilder/gossipsub/gossipSubBuilder.go

```diff
@@ -194,25 +281,31 @@ func (g *Builder) Build(ctx irrecoverable.SignalerContext) (p2p.PubSubAdapter, p
 		gossipSubConfigs.WithSubscriptionFilter(g.subscriptionFilter)
 	}
 
-	if g.rpcInspectorSuite != nil {
-		gossipSubConfigs.WithInspectorSuite(g.rpcInspectorSuite)
+	inspectorSuite, err := g.rpcInspectorSuiteFactory(
+		g.logger,
+		g.sporkId,
+		g.rpcInspectorConfig,
+		g.metricsCfg.Metrics,
+		g.metricsCfg.HeroCacheFactory,
+		g.networkType,
+		g.idProvider)
+	gossipSubConfigs.WithInspectorSuite(inspectorSuite)
```

Comment:
should this be called after the error check?

Reply:
https://github.com/onflow/flow-go/pull/4511/commits/38ed00e2f89e21065bfecb4996d3fadb9384ed63